src/components/course/data/hooks.jsx

```diff
@@ -493,6 +493,30 @@ export const useTrackSearchConversionClickHandler = ({ href, eventName }) => {
   return handleClick;
 };
 
+/**
+ * Returns a function to be used as a click handler that emits an optimizely license subsidy enrollment click event.
+ *
+ * @returns Click handler function for clicks on enrollment buttons when a learner possesses a license subsidy.
+ */
+export const useOptimizelyLicenseSubsidyEnrollmentClickHandler = ({ href, courseRunKey }) => {
```

Comment:
can we await for event push like
`await pushEvent(EVENTS.LICENSE_SUBSIDY_ENROLLMENT_CLICK, { courseKey: courseRunKey });`
we would have to make pushEvent function async
and then
`if href -> redirect`

for redirecting can we use the react router API? something like `history.push(href)`

if we can reduce this to two lines then should we move this to the component?

Reply:
This hook follows the same pattern that the other event hooks in this file are based on. Please see useOptimizelyEnrollmentClickHandler and useTrackSearchConversionClickHandler.

Reply:
strange